boards/components/src/process_console.rs

```diff
@@ -40,6 +40,10 @@ macro_rules! process_console_component_static {
         let read_buffer = kernel::static_buf!([u8; capsules::process_console::READ_BUF_LEN]);
         let queue_buffer = kernel::static_buf!([u8; capsules::process_console::QUEUE_BUF_LEN]);
         let command_buffer = kernel::static_buf!([u8; capsules::process_console::COMMAND_BUF_LEN]);
+        let command_history_buffer = kernel::static_buf!(
```

Comment:
Add two branches for this macro, one that uses the default value from the Process Console capsule and another one that allows the user to specify the length and uses. You can see a similar approach

https://github.com/UPB-CS-OpenSourceUpstream/tock/blob/fd669565a407dd03018a8acdbfe6de1a885ee912/boards/components/src/console.rs#L34-L50

Reply:
Instead of duplicating code, instead of 

```rust
() => {{ 
         use capsules::virtual_uart::MuxUart; 
         use kernel::static_buf; 
         let UART_MUX = static_buf!(MuxUart<'static>); 
         let RX_BUF = static_buf!([u8; capsules::virtual_uart::RX_BUF_LEN]); 
         (UART_MUX, RX_BUF) 
     }}; 
```

use the other branch of this

```rust
use capsules::virtual_uart::MuxUart; 
         $crate::uart_mux_component_static! (capsules::virtual_uart::RX_BUF_LEN]))
```

Reply:
Do you want suggest to add a new branch to the macro in order for the user to specify his/her COMMAND_HISTORY_LEN?

Reply:
Yes

Reply:
But is this possible? because the ProcessConsole defines the history as [Command; DEFAULT_COMMAND_HISTORY_LEN], in order to make this change possible should I add a const value in the ProcessConsole structure? Or is there a smarter mechanism that implies that DEFAULT_COMMAND_HISTORY_LEN should be equal to the USER_COMMAND_HISTORY_LEN? Thanks for answer

Reply:
I though of other one possible implementation, to make the ProcessConsole a const generics structure with COMMAND_HISTORY_LEN as the generic constant, what do you think about this implementation?

Reply:
Or to create a wrapper over the comand_history buffer with a generic const structure

Reply:
Or to introduce a new variable in the ProcessConsole struct

Reply:
I implemented the suggestion by looking at the uart_mux_component, however I could not avoid code duplication in the macro branch. Also I modified the code just in the nrf52840dk and tested it and seems to work, if the implementation is good I will modify the rest of the boards to match the changes

Reply:
Please reuse the next branch of the macro instead of writing the code again here `$crate::process_console_component_static!(..., capsules::process_console::DEFAULT_COMMAND_HISTORY_LEN)`.

Reply:
Done